Libs/MRML/Core/vtkMRMLSegmentationDisplayNode.cxx

```diff
@@ -697,25 +759,26 @@ void vtkMRMLSegmentationDisplayNode::SetSegmentDisplayPropertiesToDefault(const
 
   int wasModifyingDisplayNode = this->StartModify();
 
-  // Set segment color for merged labelmap
-  double defaultColor[3] = { 0.0, 0.0, 0.0 };
-  segment->GetDefaultColor(defaultColor);
+  // Set segment color to a default if invalid (empty)
+  double color[3] = { 0.0, 0.0, 0.0 };
+  segment->GetColor(color);
   // Generate color if default color is the default gray
-  this->NumberOfAddedSegments += 1;
-  bool generateNewDefaultColor =
-    (defaultColor[0] == vtkSegment::SEGMENT_COLOR_VALUE_INVALID[0]
-    && defaultColor[1] == vtkSegment::SEGMENT_COLOR_VALUE_INVALID[1]
-    && defaultColor[2] == vtkSegment::SEGMENT_COLOR_VALUE_INVALID[2]);
-  if (generateNewDefaultColor)
+  bool generateNewColor = ( color[0] == vtkSegment::SEGMENT_COLOR_INVALID[0]
+                         && color[1] == vtkSegment::SEGMENT_COLOR_INVALID[1]
+                         && color[2] == vtkSegment::SEGMENT_COLOR_INVALID[2] );
+  if (generateNewColor)
     {
-    this->GenerateSegmentColor(defaultColor);
+    this->GenerateSegmentColor(color);
+
+    // Set color to segment (no override is specified by default, so segment color is used)
+    segment->SetColor(color);
```

Comment:
Yes, I agree that we should commit this as is and improve in the next iteration.